Assets/MixedRealityToolkit/Definitions/Devices/MixedRealityInteractionMapping.cs

```diff
@@ -401,22 +401,9 @@ public Vector2 Vector2Data
                     Debug.LogError($"SetVector2Value is only valid for AxisType.DualAxis InteractionMappings\nPlease check the {inputType} mapping for the current controller");
                 }
 
-                if (invertXAxis || invertYAxis)
-                {
-                    float invertXAxisFactor = invertXAxis ? -1f : 1f;
-                    float invertYAxisFactor = invertYAxis ? -1f : 1f;
-
-                    Changed = !vector2Data.x.Equals(value.x * invertXAxisFactor) &&
-                              !vector2Data.y.Equals(value.y * invertYAxisFactor);
-
-                    vector2Data.x = value.x * invertXAxisFactor;
-                    vector2Data.y = value.y * invertYAxisFactor;
-                }
-                else
-                {
-                    Changed = vector2Data != value;
-                    vector2Data = value;
-                }
+                Vector2 newValue = value * new Vector2(invertXAxis ? -1f : 1f, invertYAxis ? -1f : 1f);
```

Comment:
Look mom, not more allocations!
```suggestion
                Vector2 newValue = value * (Vector2.one * (invertXAxis ? -1f : 1f, invertYAxis ? -1f : 1f));
```

Reply:
I'm not sure this suggestion compiles.
![image](https://user-images.githubusercontent.com/3580640/56942854-af1af780-6ad1-11e9-8198-27ce06c54734.png)

It doesn't appear better than using new:
![image](https://user-images.githubusercontent.com/3580640/56942831-9874a080-6ad1-11e9-9290-21505c776470.png)

Explanation of the above:
I ran three variations 1,000,000 times per update loop.
1. **TestOneMult** is a simplified, but compilable, version of the above suggestion: `var test = Vector2.one * 1;`.
2. **TestOne** is...`var test = Vector2.one;`.
3. **TestNew** is `var test = new Vector2(1, 1);`

| | TestOneMult| TestOne| TestNew|
| ------| ------------- |-------------| -----|
| Time (ms) | 494.19      | 171.82 | 161.86 |
|GC Alloc | 0     | 0|  0 |

Reply:
Those results are very curious @keveleigh , can you share the unit test you wrote them in.
Especially the .One vs new result as .One is a static constant in the struct

Reply:
Agreed. We have no way to validate your results.

>"Here guys! take these results and believe everything I tell you"

Not very scientific method of you mate. Easy way to win over someone who takes things at face value.

Reply:
I pasted the tests above exactly as they're written. I thought that was interesting as well. When you're measuring a difference of 10ms over the span of a million runs, there may be some unexpected results with rounding and whatnot. I don't really think this test says anything specific on the comparisons between those two.
Testing only 1000 times led to One being faster than New _most_ of the time.

![image](https://user-images.githubusercontent.com/3580640/56978945-84f72300-6b2d-11e9-91f9-c1a2ea25bafd.png)

![image](https://user-images.githubusercontent.com/3580640/56978002-609a4700-6b2b-11e9-8a12-2ecbe0287f91.png)

Reply:
If you've run these tests and see different results, let me know.

Reply:
>Not very scientific method of you mate. Easy way to win over someone who takes things at face value.

I posted the exact tests I ran in my original post:

![image](https://user-images.githubusercontent.com/3580640/56979668-2e8ae400-6b2f-11e9-8633-de856bf3a949.png)

Reply:
Agreed, without being able to see the test setup we can very well assume you've rigged the results in the favor of your expected result. 

Also the commit suggested was supposed to be a play on "Look Mom, no hands!" 
I'm sad no one laughed.

Reply:
Sure, but not the exact text procedure, which is important. Ask any scientist.

Reply:
Also posting a picture of the code makes it a lot harder to copy and test locally as well.

Seems like you're intentionally trying to make it difficult to validate imho.

Reply:
Here's a gist of the exact script I ran with the deep Unity profiler attached.

https://gist.github.com/keveleigh/05c22aa9c6609841636f40b823c791c6

Here's a second script I just ran to measure the methods without the profiler (I originally attached to profiler to try to detect allocations):

https://gist.github.com/keveleigh/e2ecd3b4a07c68397a6d66138678ca25

Please feel free to run both of this scripts and report back if your data appears different.

Second script data (still run 1 million times, but measured with stopwatch instead of Unity deep profiler):

| | TestOneMult| TestOne| TestNew|
| ------| ------------- |-------------| -----|
| Time (ms) | 74.3      | 26.76 | 25.04 |

Reply:
Vector2.one just returns a new Vector2. This is from source decompiled with Resharper:

```
    /// <summary>
    ///   <para>Shorthand for writing Vector2(1, 1).</para>
    /// </summary>
    public static Vector2 one
    {
      get
      {
        return new Vector2(1f, 1f);
      }
    }
```

Reply:
Oh man, I thought Unity was doing the right thing and cashing that value like they do in their other static properties. They do it for `Vector3` so I assumed it also was the same for `Vector2`. I'd like to see the results of one that's using the cached value.

Reply:
Actually, looking at the source, they do, in fact use a cached field value:

```C#
    public static Vector2 one
    {
      get
      {
        return Vector2.oneVector;
      }
    }
```

Reply:
The decompiled code is from 2016 so it must be out of date. Directional vectors got changed, too. Pretty strange that a cached field value is consistently slower in the tests.

Reply:
Haven't had a chance to test definitively and get some results. I'll post my results when I get some time later today.

Reply:
@keveleigh wow! I can't believe `new` is faster! I wonder what's even the point of using the static accessors, as I always was taught that's the benefit of even having them. 😆 

But I did add one more test to the mix. Caching that Vector2.one value locally:
```C#
[Test]
public void Test_04_01_01_Vector2_One()
{
    var watch = Stopwatch.StartNew();

    for (int i = 0; i < 10000000; i++)
    {
        var test = Vector2.one;
    }

    watch.Stop();
    Debug.Log($"Test Vector2.one execution Time: {watch.Elapsed.TotalMilliseconds} ms");

    watch = Stopwatch.StartNew();
    var vector2One = new Vector2(1f, 1f);

    for (int i = 0; i < 10000000; i++)
    {
        var test = vector2One;
    }

    watch.Stop();
    Debug.Log($"Test cached Vector2 execution Time: {watch.Elapsed.TotalMilliseconds} ms");

    watch = Stopwatch.StartNew();

    for (int i = 0; i < 10000000; i++)
    {
        var test = new Vector2(1f, 1f);
    }

    watch.Stop();
    Debug.Log($"Test new Vector2 execution Time: {watch.Elapsed.TotalMilliseconds} ms");
}
```
![image](https://user-images.githubusercontent.com/13334553/56994083-f09fb700-6b52-11e9-8325-169dd7ba5f91.png)

Reply:
Very interesting results. Glad we kept testing.